src/triggers/ipfs.ts

```diff
@@ -28,3 +28,29 @@ export const unpinnedTrackContent: (cidField: string, limit?: number) => Trigger
 
     return _.uniq(cids);
   };
+
+export const audioNotOnIpfs: Trigger<undefined> = async (clients: Clients) => {
+  const query = `select t.* from "${Table.processedTracks}" as t
+      where "lossyAudioIPFSHash" is null
+      and "lossyAudioURL" is not null
+      LIMIT ${process.env.QUERY_TRIGGER_BATCH_SIZE!}`
```

Comment:
i'm not sure this will work, since the addMetadataIPFSHash processor which runs before this will set the ipfs hashes to '' rather than null.

This is intentional, so that we can distinguish between tracks which haven't had their ipfs hash extracted from their known url vs tracks which don't actually have a hash in their known url.

The reason for this is so that we can preserve a property of the Spindexer that all processors should ideally be commutative, ie, it doesn't matter in which order they're run.

If these 2 processors were to run on a given track before the addMetadataIPFSHash then they would upload some files unnecessarily that are already on IPFS, so we should ensure these triggers runs only after the addMetadataIPFSHash one by triggering on '' rather than on null.

Reply:
it looks like the addMetadataIPFSHash processor only sets `metadataIPFSHash` to "", and doesn't change the audio or artwork hash. When I query the DB for lossyAudioIPFSHash = null I get the expected results but when querying for "" no results are returned?

Reply:
ah yeh you're right - i got confused the the metadata ipfs hash. i think for audio/artwork, their cid is always created at time of processedTrack creation via the platform type if it exists, so this is not an issue.

so your existing code should be fine, nvm sorry